### PR TITLE
tests/reference; Tests for reference inclusion

### PR DESCRIPTION
**Pull request by jlucovsky**
Issue: 4974

Positive and negative tests for reference inclusion in alerts.

## Ticket

If your pull request is related to a Suricata ticket, please provide
the full URL to the ticket here so this pull request can monitor
changes to the ticket status:

Redmine ticket: https://redmine.openinfosecfoundation.org/issues/4974

### CODE REVIEW

File: tests/reference-01/test.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+requires:
+  min-version: 8
+
+args:
+ - -k none --set outputs.1.eve-log.types.0.alert.reference=yes
```
Comment (catenacyber) [inline tests/reference-01/test.yaml:5]: I think we do not like numbers in `set` and need to resort to a suricata.yaml

File: tests/reference-01/test.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+checks:
+  - filter:
+      count: 21
+      match:
+        event_type: alert
+        alert.signature_id: 20166330
+        has-key: alert.references
```
Comment (catenacyber) [inline tests/reference-01/test.yaml:13]: Why not test the values of `alert.references` ?

**Review by jufajardini — COMMENTED** [review]
Agreeing with Philippe's comments, as those make the test more useful as examples for others who could want to use this :)

**Review by catenacyber — APPROVED** [review]
These tests look good to me, let's see the suricata PR

File: tests/reference-01/test.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+requires:
+  min-version: 8
+
+args:
```
Comment (inashivb) [inline tests/reference-01/test.yaml:4]: This pcap already exists in `dcerpc/zerologon`

**Review by inashivb — CHANGES_REQUESTED** [review]
Minor change requested inline

File: tests/reference-01/test.yaml (hunk 3 above)
Comment (jlucovsky) [reply tests/reference-01/test.yaml:4]: Thanks. Updated.

File: tests/reference-01/test.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      match:
+        event_type: alert
+        alert.signature_id: 20166330
+        alert.references: ["https://github.com/corelight/zerologon"]
+  - filter:
+      count: 1
+      match:
+        event_type: alert
+        alert.signature_id: 20166331
+        alert.references: ["https://github.com/corelight/zerologon"]
```
Comment (catenacyber) [inline tests/reference-01/test.yaml:21]: Fair enough, nit : Should we have another rule with multiple references ?

Comment (jlucovsky) [reply tests/reference-01/test.yaml:21]: Yes; will do in a new PR.

**Comment by jlucovsky** [comment]
Continued in #2022

**Pull request closed without merging**